etna/auto/optuna/config_sampler.py

```diff
@@ -0,0 +1,140 @@
+from typing import List
+from typing import Optional
+from typing import Set
+
+import numpy as np
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+from etna.auto.utils import config_hash
+from etna.auto.utils import retry
+
+
+class ConfigSampler(BaseSampler):
```

Comment:
By the way, why don't we use GridSampler from Optuna? We can somehow set the search_space to be `configs: List[dict]`

Reply:
Have discussed it in chat